### PR TITLE
[Kommander 2.2] Provide instructions to detach < 2.2 metallb from kommander

### PR DESCRIPTION
**Pull request by mwrightd2**
## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-86748

## Description of changes being made

Added workaround for MetalLB detachment for users who setup MetalLB prior to 2.2.

### Preview

You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-4279.s3-website-us-west-2.amazonaws.com/

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/blob/main/CONTRIBUTING.md) for more information.

### CODE REVIEW

**Review by mmcelroy91 — APPROVED** [review]
Did not find anything wrong... only question is 'helm' supposed to be capitalized?  Probably not, but had to ask.

**Comment by mwrightd2** [comment]
> Did not find anything wrong... only question is 'helm' supposed to be capitalized? Probably not, but had to ask.

I don't believe so in this context.

**Comment by gvasquezvargas** [comment]
@mackdubya since we have a PR for the upgrade copy that will live in Konvoy, should I copy this into https://github.com/mesosphere/dcos-docs-site/pull/4272?

**Comment by mwrightd2** [comment]
Yes please @gvasquezvargas

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 1]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
   ```
-  
+
   If you have any DKP Catalog Applications, download the DKP Catalog Application charts bundle:
 
   ```bash
   wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-catalog-applications-charts-bundle-v2.2.0.tar.gz"
   ```
+  
+  <p class="message--note"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a catalog application. If you installed MetalLB on     the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading..</p>
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:42]:
```suggestion
  <p class="message--note"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a catalog application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
```

**Review by joejulian — APPROVED** [review]
lgtm, but someone from Kommander should double check.

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 2]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  
+  1. Pause the helm release.
+  ```sh
```
Comment (gvasquezvargas) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:45]: I think all of these should be bash? but not sure... 
```suggestion
  ```bash
```

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 3]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
   ```
-  
+
   If you have any DKP Catalog Applications, download the DKP Catalog Application charts bundle:
 
   ```bash
   wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-catalog-applications-charts-bundle-v2.2.0.tar.gz"
   ```
+  
+  <p class="message--note"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a catalog application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:42]:
```suggestion
  <p class="message--important"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a platform application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
```

Might bump this one up to an important

Comment (gracedo) [reply pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:42]: Maybe we can include this section under a subheading to further highlight these steps?

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 4]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  
+  1. Pause the helm release.
+  ```bash
+  kubectl -n kommander patch -p='{"spec":{"suspend": true}}' --type=merge helmrelease/metallb
+  helmrelease.helm.toolkit.fluxcd.io/metallb patched
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:47]: this line is technically the output, so it should go in a separate ```sh section

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 5]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  ```
+
+  1. Delete the helm release secret.
+  ```bash
+  kubectl -n kommander delete secret -l name=metallb,owner=helm
+  secret "sh.helm.release.v1.metallb.v1" deleted
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:53]: same here, this is output

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 6]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  ```
+  1. Delete MetalLB.
+  ```bash
+  k -n kommander delete appdeployment metallb
+  appdeployment.apps.kommander.d2iq.io "metallb" deleted
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:58]: same here, this is output

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 7]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  ```
+
+  1. Unpause the helm release.
+  ```bash
+  kubectl -n kommander patch -p='{"spec":{"suspend": false}}' --type=merge helmrelease/metallb
+  helmrelease.helm.toolkit.fluxcd.io/metallb patched
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:64]: same here, this is output

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 8]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  ```
+  This deletes MetalLb from Kommander while leaving the resources running in the cluster.
+  
+  ```bash
+  $ kubectl -n kommander get pod -l app=metallb
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:69]:
```suggestion
  ```bash
  kubectl -n kommander get pod -l app=metallb
```

maybe separate out the command + sample output?

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 9]
```diff
@@ -20,25 +20,60 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  NAME                                 READY   STATUS    RESTARTS   AGE
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:70]:
```suggestion
  
  ```sh
  NAME                                 READY   STATUS    RESTARTS   AGE
```

too many ticks to suggest this properly 🤦🏻‍♀️  but essentially wrapping the kubectl cmd in bash
and then separating out the output into sh

**Comment by mwrightd2** [comment]
Made those edits @gracedo

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 10]
```diff
@@ -20,25 +20,72 @@ This section describes how to upgrade your Kommander Management cluster and all
...
   wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-kommander-charts-bundle-v2.2.0.tar.gz"
   ```
-  
+
   If you have any DKP Catalog Applications, download the DKP Catalog Application charts bundle:
 
   ```bash
   wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-catalog-applications-charts-bundle-v2.2.0.tar.gz"
   ```
+## Detatch MetalLB from Kommander
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:41]:
```suggestion
## Detach MetalLB from Kommander
```

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 11]
```diff
@@ -20,25 +20,72 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+
+  <p class="message--important"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a platform application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
+  
+  1. Pause the helm release.
+  ```bash
+  kubectl -n kommander patch -p='{"spec":{"suspend": true}}' --type=merge helmrelease/metallb
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:47]: missed a spot to separate out the input/outputs here!

File: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md [hunk 12]
```diff
@@ -20,25 +20,72 @@ This section describes how to upgrade your Kommander Management cluster and all
...
+  kubectl -n kommander patch -p='{"spec":{"suspend": false}}' --type=merge helmrelease/metallb
+  ```
+  
+  ```sh 
+  helmrelease.helm.toolkit.fluxcd.io/metallb patched
+  ```
+  This deletes MetalLb from Kommander while leaving the resources running in the cluster.
+  
+  ```bash
+  kubectl -n kommander get pod -l app=metallb
```
Comment (gracedo) [inline pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md:79]: can't suggest ticks properly 🙈 but there needs to be a closing ``` here

**Comment by mwrightd2** [comment]
Can't even spell detach smh, edits up @gracedo

**Review by nataliepina — APPROVED** [review]

**Pull request merged**